### ISSUE TITLE
Feature request: Ability to disable stretched video

### ISSUE THREAD

**Issue by rschroll** [H-M-H/Weylus#19]
In many cases, I would prefer to have the aspect ratio of the video maintained, even if that means dead space around the video.  Simply disabling the transformations on the video element is enough.  I made a simple test where the stretching could be disabled via a location hash.  Either way, all of the video is visible and the touch registration is correct.  This works both when the video is too tall and too wide, so there's no math updates needed.

The thing that's blocking me for making a PR is that I don't know how the UI should work.  Providing a separate URL, or just a different hash, for the two options would be easiest to implement, and it would make on-the-fly changes easy.  This isn't particularly user friendly.  A HTML UI would not be too hard, but it would either always take up space, or it would need some way to be triggered and then hidden.  An option in the server GUI would make sense, in that all of the other options are there already.  But this would either need a server restart to take effect, or a communication channel to the client to update the state.

I'm happy to work on a PR, if you let me know your preference.

**Comment by H-M-H** [comment]
This issue touches a pain point I have with how configuration works :) Weylus is capable to handle multiple connections/clients but client specific options like whether to capture the cursor can only be changed for all clients together and only be restarting the server.

I have given this some thought and come to the conclusion that it is probably best to move all client specific options (all the options on the right in the current gui and the screen update interval) to an HTML UI. The chosen config can then be sent over websockets (this needs some protocol changes). Stretching or not stretching the video should be handled purely in the browser.

Changing the websockets protocol to do configuration is not a problem, I am already pretty sure how I will do this. The only thing I am not sure how to do is the HTML UI. Some thoughts:
- Page reloads are out of the question as this would close the websocket connections
- I absolutely do not want to waste any space that could be used for pen/touch input for a menu
  - Only display a menu on startup and remove it after configuration, drawback is no dynamic configuration / page reload is required to change config
  - find a way to toggle a menu without touch input from the video element that works on mobile devices (not sure if that's even possible)
  - probably the best option: implement some kind of sidebar/drawer, as seen in many apps, that can be pulled in from the left.
    - something like https://mac81.github.io/pure-drawer/ but with the hamburger replaced by pulling it from the edge of the browser window.
    - if possible do not introduce javascript/css dependencies i.e. calling `npm install xyz`
- store last config with cookies to reduce friction when one just restarts Weylus to continue previous work

**Comment by rschroll** [comment]
Sounds like it's worth experimenting with a HTML interface here, where we don't have to worry about client-server communication.  I can play with some experiments.

Is there a way to run Weylus in a way where it pulls the static web files from the local filesystem, instead of using the versions baked into the binary?  That would speed up the trial and error process a lot.  If not, I can probably figure out how to stick a proxy in between Weylus and the client and serve the files that way.

**Comment by H-M-H** [comment]
>  I can play with some experiments.

Nice, I am working on the protocol changes right now.

> Is there a way to run Weylus in a way where it pulls the static web files from the local filesystem

You can just download "/", "style.css" and "lib.js" form Weylus or use https://github.com/H-M-H/Weylus/tree/master/www and https://github.com/H-M-H/Weylus/tree/master/ts and serve it with a webserver of your choice.

**Comment by H-M-H** [comment]
You can have a look at the new protocol here: https://github.com/H-M-H/Weylus/tree/html_config. It needs some more testing and polishing but I think it should be pretty easy to integrate it into an html ui now. Please let me know if you get a decent interface working.

**Comment by rschroll** [comment]
I have a PR up in #22.  It just handles the stretch setting right now, but I gave it plenty of room for other settings.  I don't know if you want to discuss it here or there.

I figured out the web server trick once I noticed the video and touch events were going over different ports.  For the record, here's a little Flask server I put together for development:  [serve.py](https://github.com/H-M-H/Weylus/files/4808763/serve.py.txt).  You just run it from the `www/` direrctory and it picks up the template and static files correctly.
